### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by necessarily-equal**
The last commit removing exceptions is by Kangz, and is needed for WASM stuff.

### CODE REVIEW

File: src/common/FileSystem.cpp [hunk 1]
```diff
@@ -2067,24 +2071,26 @@ Cmd::CompletionResult CompleteFilename(Str::StringRef prefix, Str::StringRef roo
...
+		std::string ext = Path::Extension(x);
+		if (!extension.empty() && ext != extension && !(allowSubdirs && ext == "/"))
+			continue;
+		std::string result;
+		if (stripExtension)
+			result = Path::Build(prefixDir, Path::StripExtension(x));
+		else
+			result = Path::Build(prefixDir, x);
+		if (Str::IsPrefix(prefix, result))
+		out.emplace_back(result, "");
```
Comment (Kangz) [inline src/common/FileSystem.cpp:2091]: nit:  this indentation is misleading, the `emplace_back` is done inside the if.

File: src/common/Color.h [hunk 2]
```diff
@@ -35,7 +35,6 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <type_traits>
 
 #include "Compiler.h"
-#include "Math.h"
```
Comment (Kangz) [inline src/common/Color.h:38]: It seems that this removal is making the appveyor fail.

File: src/common/FileSystem.cpp [hunk 3]
```diff
@@ -2067,24 +2071,26 @@ Cmd::CompletionResult CompleteFilename(Str::StringRef prefix, Str::StringRef roo
...
-				result = Path::Build(prefixDir, Path::StripExtension(x));
-			else
-				result = Path::Build(prefixDir, x);
-			if (Str::IsPrefix(prefix, result))
-				out.emplace_back(result, "");
-		}
-		return out;
-	} catch (std::system_error&) {
+	std::error_code err;
+	DirectoryRange range = HomePath::ListFiles(Path::Build(root, prefixDir));
```
Comment (slipher) [inline src/common/FileSystem.cpp:2075]: forgot to put in `err`

File: src/common/Color.h (hunk 2 above)
Comment (necessarily-equal) [reply src/common/Color.h]: Looks like line 272 uses math stuff. No idea why it works with GCC.

**Review by Kangz — APPROVED** [review]

**Comment by illwieckz** [comment]
Wow, you deleted some TTimo's comment mentionning Wolf MP? That's Sacrilegious! What would you do next time, fix the typo in the comment about the QuakeEdRadient editor? People today don't have respect for history and traditions! 🤣

**Pull request merged**